mmdet/apis/inference.py

```diff
@@ -59,45 +68,19 @@ def inference_detector(model, imgs):
         detection results directly.
     """
     cfg = model.cfg
-    img_transform = ImageTransform(
-        size_divisor=cfg.data.test.size_divisor, **cfg.img_norm_cfg)
-
     device = next(model.parameters()).device  # model device
-    if not isinstance(imgs, list):
-        return _inference_single(model, imgs, img_transform, device)
-    else:
-        return _inference_generator(model, imgs, img_transform, device)
-
-
-def _prepare_data(img, img_transform, cfg, device):
-    ori_shape = img.shape
-    img, img_shape, pad_shape, scale_factor = img_transform(
-        img,
-        scale=cfg.data.test.img_scale,
-        keep_ratio=cfg.data.test.get('resize_keep_ratio', True))
-    img = to_tensor(img).to(device).unsqueeze(0)
-    img_meta = [
-        dict(
-            ori_shape=ori_shape,
-            img_shape=img_shape,
-            pad_shape=pad_shape,
-            scale_factor=scale_factor,
-            flip=False)
-    ]
-    return dict(img=[img], img_meta=[img_meta])
-
-
-def _inference_single(model, img, img_transform, device):
-    img = mmcv.imread(img)
-    data = _prepare_data(img, img_transform, model.cfg, device)
+    # build the data pipeline
+    test_pipeline = [LoadImage()] + cfg.data.test.pipeline[1:]
+    test_pipeline = Compose(test_pipeline)
+    # prepare data
+    data = dict(img=img)
```

Comment:
There's a regression here -- previously, `inference_detector` supported multiple paths, not just single path. Documentation (pydoc) still says it supports multiple (e.g. https://github.com/open-mmlab/mmdetection/pull/935/files#diff-f3f6abcbd301661a8751f22dd131f666R64), but it actually fails when trying to read the list with `imread`